plugins/node/opentelemetry-instrumentation-aws-lambda/src/instrumentation.ts

```diff
@@ -97,14 +97,28 @@ export class AwsLambdaInstrumentation extends InstrumentationBase<AwsLambdaInstr
     // Lambda loads user function using an absolute path.
     let filename = path.resolve(taskRoot, moduleRoot, module);
     if (!filename.endsWith('.js')) {
-      // its impossible to know in advance if the user has a cjs or js file.
-      // check that the .js file exists otherwise fallback to next known possibility
+      // Its impossible to know in advance if the user has a js, cjs or mjs file.
+      // Check that the .js file exists otherwise fallback to the next known possibilities (.cjs, .mjs).
```

Comment:
Note that the Lambda Node.js runtime prefers this order: .js, .mjs, .cjs
(https://github.com/aws/aws-lambda-nodejs-runtime-interface-client/blob/v3.2.1/src/UserFunction.js#L149-L174)
I think this means that if a user has a Lambda with both a 'handler.mjs' *and* 'handler.cjs' that instrumentation won't work: OTel will prepare to instrument the .cjs file, but the Lambda runtime will load the .mjs file.

I think this is definitely an improvement, though, so if we wanted to get this in *now* for a release it is good.

Reply:
Good catch

Reply:
done